cartridges/docker/apache-hbase-1.0.1.1/plugins/hbase-topology-handler.py

```diff
@@ -0,0 +1,167 @@
+
+            self.remove_data_from_metadata("CONFIG_PARAM_HBASE_MASTER_HOSTNAME")
+            self.remove_data_from_metadata("CONFIG_PARAM_HBASE_REGIONSERVER_DATA")
+
+            mdsclient.MDSPutRequest()
+            master_hostname=socket.gethostname()
+            master_hostname_property = {"key": "CONFIG_PARAM_HBASE_MASTER_HOSTNAME", "values": [ master_hostname ]}
+            mdsclient.put(master_hostname_property, app=True)
+            os.environ["CONFIG_PARAM_HBASE_MASTER_HOSTNAME"] = master_hostname
+            log.info("Published CONFIG_PARAM_HBASE_MASTER_HOSTNAME to metadata")
+            log.info(master_hostname)
+
+        else:
+
+            mds_response = None
+            while mds_response is None:
+                log.debug("Waiting for CONFIG_PARAM_HBASE_MASTER_HOSTNAME to be available from metadata service for app ID: %s" % values["APPLICATION_ID"])
+                time.sleep(5)
+                mds_response = mdsclient.get(app=True)
+                if mds_response is not None and mds_response.properties.get("CONFIG_PARAM_HBASE_MASTER_HOSTNAME") is None:
+                    mds_response = None
+
+            master_hostname = mds_response.properties["CONFIG_PARAM_HBASE_MASTER_HOSTNAME"]
+            os.environ["CONFIG_PARAM_HBASE_MASTER_HOSTNAME"] = master_hostname
+            log.info("Received metadata for CONFIG_PARAM_HBASE_MASTER_HOSTNAME")
+            log.info(master_hostname)
+
+            config_command = "echo "+hbase_master_ip+"    "+master_hostname+"  >> /etc/hosts"
+            env_var = os.environ.copy()
+            p = subprocess.Popen(config_command, env=env_var, shell=True)
+            output, errors = p.communicate()
+            log.info("Entry added to /etc/hosts")
+
+            server_hostname=socket.gethostname()
+            server_ip=socket.gethostbyname(server_hostname)
+
+            add_host_command = "ssh root@"+hbase_master_ip+" 'bash -s' < /tmp/add-host.sh "+server_ip+" "+server_hostname
+            env_var = os.environ.copy()
+            p = subprocess.Popen(add_host_command, env=env_var, shell=True)
+            output, errors = p.communicate()
+            log.info("Entry added to Hbase master /etc/hosts")
+
+            mdsclient.MDSPutRequest()
+            data = server_hostname+":"+server_ip
+            regionserver_data_property = {"key": "CONFIG_PARAM_HBASE_REGIONSERVER_DATA", "values": [ data ]}
+            mdsclient.put(regionserver_data_property, app=True)
+            log.info("Published CONFIG_PARAM_HBASE_REGIONSERVER_DATA to metadata")
+            log.info(data)
+
+
+        # configure server
+        log.info("Configuring HBase...")
+        config_command = "python ${CONFIGURATOR_HOME}/configurator.py"
+        env_var = os.environ.copy()
+        p = subprocess.Popen(config_command, env=env_var, shell=True)
+        output, errors = p.communicate()
+        log.info("HBase configured successfully")
+        os.environ["CONFIGURED"] = "true"
```

Comment:
This environment variable is using to check whether the configuration completion is  done before executing the startup commands. Otherwise instance will start without complete configuration changes.